tasks/deploy/config/mainnet-fork/steth-30day.ts

```diff
@@ -0,0 +1,74 @@
+import { formatEther, parseEther } from "viem";
+import {
+    HyperdriveInstanceConfig,
+    MAINNET_STETH_ADDRESS,
+    getLinkerDetails,
+    normalizeFee,
+    parseDuration,
+    toBytes32,
+} from "../../lib";
+
+const CONTRIBUTION = parseEther("500");
+
+export const MAINNET_FORK_STETH_30DAY: HyperdriveInstanceConfig<"StETH"> = {
+    name: "STETH_30_DAY",
+    prefix: "StETH",
+    coordinatorAddress: async (hre) =>
+        hre.hyperdriveDeploy.deployments.byName("STETH_COORDINATOR").address,
+    deploymentId: toBytes32("STETH_30_DAY"),
+    salt: toBytes32("0xababe"),
+    extraData: "0x",
+    contribution: CONTRIBUTION,
+    fixedAPR: parseEther("0.05"),
+    timestretchAPR: parseEther("0.05"),
+    options: {
+        destination: process.env.ADMIN! as `0x${string}`,
+        asBase: false,
+        extraData: "0x",
+    },
+    prepare: async (hre) => {
+        // mint steth
+        await hre.run("fork:mint-steth", {
+            amount: formatEther(CONTRIBUTION),
+            address: (await hre.getNamedAccounts())["deployer"],
+        });
+        let vaultSharesToken = await hre.viem.getContractAt(
+            "MockLido",
+            MAINNET_STETH_ADDRESS,
+        );
+        let pc = await hre.viem.getPublicClient();
+        // approve the coordinator
+        // TODO: Investigate why significant additional allowance is needed for deposit (10 ether is too little)
```

Comment:
I think that what's going on here is that steth approvals are in units of tokens rather than shares: https://github.com/lidofinance/lido-dao/blob/master/contracts/0.4.24/StETH.sol#L212, but we are trying to deposit `500.001e18 shares`. Let's say that the share price is `1.2e18 steth/share`. Then we'd need an approval amount of  `600.0012 steth`.

Reply:
We should be able to calculate this close to perfectly using `getPooledEthByShares`.

Reply:
fixed